src/richie/apps/core/context_processors.py

```diff
@@ -86,3 +92,88 @@ def site_metas(request):
 
     context["FRONTEND_CONTEXT"] = json.dumps(context["FRONTEND_CONTEXT"])
     return context
+
+
+class WebAnalyticsContextProcessor:
+    """
+    Context processor to add Web Analytics tracking information to Richie CMS templates and
+    frontend.
+    """
+
+    def context_processor(self, request: HttpRequest) -> dict:
+        """
+        Real implementation of the context processor for the Web Analytics core app sub-module
+        """
+        context = {}
+        if hasattr(request, "current_page"):
+            # load web analytics settings to the context
+            if getattr(settings, "WEB_ANALYTICS_ID", None):
+                context["WEB_ANALYTICS_ID"] = settings.WEB_ANALYTICS_ID
+            context["WEB_ANALYTICS_LOCATION"] = getattr(
+                settings, "WEB_ANALYTICS_LOCATION", "head"
+            )
+            context["WEB_ANALYTICS_PROVIDER"] = getattr(
+                settings, "WEB_ANALYTICS_PROVIDER", "google_analytics"
+            )
+            if getattr(settings, "WEB_ANALYTICS_ID", None):
+                context["WEB_ANALYTICS_DIMENSIONS"] = self.get_dimensions(request)
```

Comment:
You can now group these 2 :wink: 
```suggestion
            if getattr(settings, "WEB_ANALYTICS_ID", None):
                context["WEB_ANALYTICS_ID"] = settings.WEB_ANALYTICS_ID
                context["WEB_ANALYTICS_DIMENSIONS"] = self.get_dimensions(request)
            context["WEB_ANALYTICS_LOCATION"] = getattr(
                settings, "WEB_ANALYTICS_LOCATION", "head"
            )
            context["WEB_ANALYTICS_PROVIDER"] = getattr(
                settings, "WEB_ANALYTICS_PROVIDER", "google_analytics"
            )
```

Reply:
Done